src/modules/imaging/location.py

```diff
@@ -0,0 +1,178 @@
+
+    def location(self) -> LatLng:
+        return self._current_location
+
+    def heading(self) -> Heading:
+        return self._current_heading
+
+    def altitude(self) -> float:
+        return self._current_altitude
+
+    def orientation(self) -> Rotation:
+        return self._current_orientation
+
+    def set_location(self, new_location):
+        self._current_location = new_location
+
+    def set_heading(self, new_heading):
+        self._current_heading = new_heading
+
+    def set_altitude(self, new_altitude):
+        self._current_altitude = new_altitude
+
+    def set_orientation(self, new_orientation):
+        self._current_orientation = new_orientation
+
+    def debug_change_location(self, **kwargs):
+        """
+        Change the location reported by this DebugLocationProvider.
+
+        Keyword Arguments:
+        lat -- Latitude
+        lng -- Longitude
+        heading -- Heading direction
+        altitude -- Altitude
+        pitch -- Pitch angle
+        roll -- Roll angle
+        yaw -- Yaw angle
+        """
+        if 'lat' in kwargs and 'lng' in kwargs:
+            self._current_location = LatLng(kwargs['lat'], kwargs['lng'])
+
+        if 'heading' in kwargs:
+            self._current_heading = Heading(kwargs['heading'])
+
+        if 'altitude' in kwargs:
+            self._current_altitude = kwargs['altitude']
+
+        if all(k in kwargs for k in ['pitch', 'roll', 'yaw']):
+            self._current_orientation = Rotation(kwargs['pitch'],
+                                                 kwargs['roll'], kwargs['yaw'])
+
+
+class MAVLinkLocationProvider:
+    """
+    Will provide location information based on information received as MAVLink messages.
+    """
+
+    def __init__(self, mavlink_delegate: MAVLinkDelegate):
+        self.mavlink_delegate = mavlink_delegate
+        self._location = None
+        self._heading = None
+        self._altitude = None
+        self._orientation = None
+
+        # Subscribe to the delegate's messages
+        self.mavlink_delegate.subscribe(self._process_message)
+
+    def _process_message(self, message):
+        # This callback processes incoming MAVLink messages and updates the internal state
+        if message.get_type() == 'GLOBAL_POSITION_INT':
+            self._location = LatLng(message.lat / 1e7, message.lon / 1e7)
+            self._altitude = message.alt / 1000.0  # Altitude in meters
+            self._heading = Heading(message.hdg / 1e7)  # Heading in degrees
+        elif message.get_type() == 'ATTITUDE':
+            self._orientation = Rotation(pitch=math.degrees(message.pitch),
+                                         roll=math.degrees(message.roll),
+                                         yaw=math.degrees(message.yaw))
+
+    def location(self) -> LatLng:
+        if self._location is not None:
+            return self._location
+        else:
+            raise ValueError("No valid location data available")
```

Comment:
I'll add a test that we raise a `ValueError` if the `MAVLinkLocationProvider` has not yet received a `location`/`heading`/...